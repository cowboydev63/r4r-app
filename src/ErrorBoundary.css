.r4r-boundary {
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background: rgb(226, 226, 226);
}

.boundary__inner {
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;
    align-items: center;
    background: white;
    box-shadow: 0 15px 15px -12px rgba(0,0,0,0.3);
    padding: 15px;
}

.r4r-boundary p {
    margin: 10px;
}