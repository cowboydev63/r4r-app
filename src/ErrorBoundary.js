import React from 'react';
import PropTypes from 'prop-types';
import Error from './components/Error';
import { connect } from 'react-redux';

export class ErrorBoundary extends React.Component {
    static propTypes = {
        error: PropTypes.string,
        url: PropTypes.string.isRequired,
        children: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.node),
            PropTypes.node
        ]),
        location: PropTypes.shape({
            search: PropTypes.string.isRequired,
        })
    }
    
    render(){
        if(this.props.error){
            return <Error message={ this.props.error }/>
        }
        return this.props.children;
    }
}

const mapStateToProps = ({ error, router }) => ({
    error,
    location: router.location,
})

export default connect(mapStateToProps)(ErrorBoundary);