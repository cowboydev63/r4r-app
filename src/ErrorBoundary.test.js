import React from 'react';
import { shallow } from 'enzyme';
import { ErrorBoundary } from './ErrorBoundary';

describe('Error Boundary', ()=> {
  it('renders without crashing', () => {
    shallow(<ErrorBoundary />);
  });

})