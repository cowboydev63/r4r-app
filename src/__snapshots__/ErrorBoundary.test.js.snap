// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary HOC renders correctly 1`] = `""`;
